Ignore usage of DOM related typescript type or interface and lint react component that returns fragments

vietanhtran16 opened the pull request.

vietanhtran16 committed:
Rename package

vietanhtran16 committed:
Do not check useage of DOM related interface

vietanhtran16 committed:
Add unit tests to cover scenario that typescript type and interface should be ignored

jakubriedl commented on package.json:
```diff
@@ -1,5 +1,5 @@
 {
-  "name": "eslint-plugin-ssr-friendly",
+  "name": "@cultureamp/eslint-plugin-ssr-friendly",
```
you shouldn't rename package when it is fork. We shouldn't need to publish it to registery and just install from git
```suggestion
  "name": "eslint-plugin-ssr-friendly",
```

vietanhtran16 replied:
I have put it back now

jakubriedl commented on package.json:
```diff
@@ -1,5 +1,5 @@
 {
-  "name": "eslint-plugin-ssr-friendly",
+  "name": "@cultureamp/eslint-plugin-ssr-friendly",
   "version": "1.0.5",
```
```suggestion
  "version": "1.0.6",
```

vietanhtran16 replied:
good catch

vietanhtran16 committed:
Dont rename package

vietanhtran16 committed:
v1.0.6

jakubriedl commented on package-lock.json:
```diff
@@ -1,8 +1,2638 @@
 {
```
this package-lock doesn't seem right, it  contain the wrong name and version and also adds ridiculous amount of lines. Isn't that caused by some link or something? can we cleanup the changes here?

vietanhtran16 replied:
hmm 🤔 that is true. I will check and clean it up

vietanhtran16 committed:
Detects unsafe DOM in react component that return fragment

vietanhtran16 committed:
Clean up package-lock

vietanhtran16 committed:
Add typescript

vietanhtran16 committed:
Remove redundant code

vietanhtran16 commented:
Hey @jakubriedl, I have updated the PR accordingly. I have also added a fix where it should detects errors in component that return fragments (that is why it did not pick up errors previously on VideoPlayer.tsx in kaizen since it returned a fragment)
